java-checks/src/main/java/org/sonar/java/checks/ReplaceLambdaByMethodRefCheck.java

```diff
@@ -155,7 +155,21 @@ private static boolean isMethodInvocation(@Nullable Tree tree, LambdaExpressionT
         }
         arguments = ((NewClassTree) tree).arguments();
       } else {
-        arguments = ((MethodInvocationTree) tree).arguments();
+        MethodInvocationTree mit = (MethodInvocationTree) tree;
+        if(mit.methodSelect().is(Tree.Kind.MEMBER_SELECT)) {
+          MemberSelectExpressionTree mse = (MemberSelectExpressionTree) mit.methodSelect();
+          while (mse != null) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS134)